spec/models/person_spec.rb

```diff
@@ -210,4 +210,283 @@
+      admin_user.tags.clear
+      admin_user.save!
+    end
+
+    it "allow person creation only with admin tags" do
+      person_tag1 = create(:person_tag)
+      person_tag2 = create(:alt_person_tag)
+
+      admin_user.tags << person_tag1
+      admin_user.save!
+
+      expect do
+        person1 = Person.new
+        person1.tags << person_tag1
+        person1.save!
+      end.to change { Person.count }.by(1)
+
+      expect do
+        person2 = Person.new
+        person2.tags << person_tag2
+        expect(person2).to_not be_valid
+        expect(person2.errors[:person]).to eq(['Person tags not allowed'])
+      end.to change { Person.count }.by(0)
+
+      admin_user.tags << person_tag2
+      admin_user.save!
+
+      expect do
+        person3 = Person.new
+        person3.tags << person_tag1
+        person3.save!
+      end.to change { Person.count }.by(1)
+
+      expect do
+        person4 = Person.new
+        person4.tags << person_tag2
+        person4.save!
+      end.to change { Person.count }.by(1)
+    end
+
+    it "allow person creation with tags if admin has no tags" do
+      person_tag1 = create(:person_tag)
+      person_tag2 = create(:alt_person_tag)
+
+      expect do
+        person1 = Person.new
+        person1.tags << person_tag1
+        person1.save!
+      end.to change { Person.count }.by(1)
+
+      expect do
+        person2 = Person.new
+        person2.tags << person_tag2
+        person2.save!
+      end.to change { Person.count }.by(1)
+    end
+
+    it "allow person creation without tags if admin has no tags" do
+      expect do
+        person = Person.new
+        person.save!
+      end.to change { Person.count }.by(1)
+    end
+
+    it "allow person creation without tags if admin has tags" do
+      person_tag = create(:person_tag)
+
+      admin_user.tags << person_tag
+      admin_user.save!
+
+      expect do
+        person = Person.new
+        person.save!
+      end.to change { Person.count }.by(1)
+    end
+
+    it "allow change state with person tags if admin has tags" do
+      person1, person2, person3, person4 = setup_for_admin_tags_spec
+
+      admin_user.tags << person1.tags.first
+      admin_user.save!
+
+      %i{enable disable reject}.each do |action|
+        person = Person.find(person1.id)
+        person.send("#{action}!")
+
+        person = Person.find(person2.id)
+        person.send("#{action}!")
+
+        expect { Person.find(person3.id) }.to raise_error(ActiveRecord::RecordNotFound)
+
+        person = Person.find(person4.id)
+        person.send("#{action}!")
+      end
+
+      admin_user.tags << person3.tags.first
+      admin_user.save!
+
+      %i{enable disable reject}.each do |action|
+        person = Person.find(person3.id)
+        person.send("#{action}!")
+      end
+    end
+
+    it "Update a person with tags if admin has tags" do
+      person1, person2, person3, person4 = setup_for_admin_tags_spec
+
+      person = Person.find(person1.id)
```

Comment:
Este test está muy verboso. Por que se hace el find por cada ID y luego el enabled y save?
```suggestion
      person1, person2, person3, person4 = people = setup_for_admin_tags_spec
      people.each{|i| i.reload.update!(enabled: false) }
```